File: Cardiovascular Disease Prediction using Machine Learning and Clinical Data/scripts/python/statistical_analysis.py
"""
Statistical Analysis: Descriptive, Inferential, and Exploratory

This script performs comprehensive statistical analysis of the Cardiovascular Disease Dataset,
including descriptive statistics, inferential statistics, and exploratory data analysis.
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from scipy.stats import ttest_ind, chi2_contingency, f_oneway, shapiro
import warnings
warnings.filterwarnings('ignore')

# Set style
sns.set_style("whitegrid")
plt.rcParams['figure.figsize'] = (12, 6)

def load_data():
    """Load the dataset"""
    df = pd.read_csv('../../data/Cardiovascular_Disease_Dataset.csv')
    print("Dataset loaded successfully!")
    print(f"Shape: {df.shape}")
    return df

def descriptive_statistics(df):
    """Perform descriptive statistics"""
    print("=" * 80)
    print("DESCRIPTIVE STATISTICS")
    print("=" * 80)
    
    numerical_cols = ['age', 'restingBP', 'serumcholestrol', 'maxheartrate', 'oldpeak']
    categorical_cols = ['gender', 'chestpain', 'fastingbloodsugar', 'restingrelectro', 
                        'exerciseangia', 'slope', 'noofmajorvessels', 'target']
    
    print("\nNumerical Variables Summary:")
    print(df[numerical_cols].describe())
    
    print("\n\nCategorical Variables Summary:")
    for col in categorical_cols:
        if col in df.columns:
            print(f"\n{col.upper()}:")
            print(df[col].value_counts())
            print(f"Percentage: {df[col].value_counts(normalize=True) * 100}")

def inferential_statistics(df):
    """Perform inferential statistics"""
    print("=" * 80)
    print("INFERENTIAL STATISTICS")
    print("=" * 80)
    
    numerical_cols = ['age', 'restingBP', 'serumcholestrol', 'maxheartrate', 'oldpeak']
    disease_0 = df[df['target'] == 0]
    disease_1 = df[df['target'] == 1]
    
    print("\nT-Test: Comparing means between disease and no-disease groups")
    for col in numerical_cols:
        if col in df.columns:
            group_0 = disease_0[col].dropna()
            group_1 = disease_1[col].dropna()
            
            if len(group_0) > 0 and len(group_1) > 0:
                statistic, p_value = ttest_ind(group_0, group_1)
                print(f"\n{col.upper()}:")
                print(f"  t-statistic: {statistic:.4f}")
                print(f"  p-value: {p_value:.4f}")
                print(f"  Significant: {'Yes' if p_value < 0.05 else 'No'}")

def exploratory_analysis(df):
    """Perform exploratory data analysis"""
    print("=" * 80)
    print("EXPLORATORY DATA ANALYSIS")
    print("=" * 80)
    
    # Correlation matrix
    numerical_cols = ['age', 'restingBP', 'serumcholestrol', 'maxheartrate', 'oldpeak', 'target']
    correlation_matrix = df[numerical_cols].corr()
    
    plt.figure(figsize=(10, 8))
    sns.heatmap(correlation_matrix, annot=True, fmt='.2f', cmap='coolwarm', center=0)
    plt.title('Correlation Matrix')
    plt.tight_layout()
    plt.savefig('../../results/correlation_matrix.png')
    plt.close()
    
    print("\nCorrelation with Target:")
    print(correlation_matrix['target'].sort_values(ascending=False))

def main():
    """Main function"""
    df = load_data()
    descriptive_statistics(df)
    inferential_statistics(df)
    exploratory_analysis(df)
    print("\nAnalysis complete!")

if __name__ == "__main__":
    main()

File: Cardiovascular Disease Prediction using Machine Learning and Clinical Data/scripts/r/statistical_analysis.R
# Statistical Analysis: Descriptive, Inferential, and Exploratory
#
# This script performs comprehensive statistical analysis of the Cardiovascular Disease Dataset,
# including descriptive statistics, inferential statistics, and exploratory data analysis.

# Load necessary libraries
library(tidyverse)
library(ggplot2)
library(dplyr)
library(corrplot)
if (!require(psych)) install.packages("psych")
if (!require(Hmisc)) install.packages("Hmisc")
library(psych)
library(Hmisc)

# Set theme for plots
theme_set(theme_minimal() + theme(plot.title = element_text(size = 14, face = "bold")))

# Load the dataset
df <- read.csv("../../data/Cardiovascular_Disease_Dataset.csv", stringsAsFactors = FALSE)

cat("Dataset loaded successfully!\n")
cat("Shape:", nrow(df), "rows,", ncol(df), "columns\n")

# Descriptive Statistics
cat(paste(rep("=", 80), collapse = ""), "\n")
cat("DESCRIPTIVE STATISTICS\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

numerical_cols <- c("age", "restingBP", "serumcholestrol", "maxheartrate", "oldpeak")
categorical_cols <- c("gender", "chestpain", "fastingbloodsugar", "restingrelectro", 
                     "exerciseangia", "slope", "noofmajorvessels", "target")

cat("\nNumerical Variables Summary:\n")
print(summary(df[numerical_cols]))

cat("\n\nCategorical Variables Summary:\n")
for (col in categorical_cols) {
  if (col %in% names(df)) {
    cat("\n", toupper(col), ":\n")
    print(table(df[[col]]))
    cat("Percentage:\n")
    print(prop.table(table(df[[col]])) * 100)
  }
}

# Inferential Statistics
cat("\n\n", paste(rep("=", 80), collapse = ""), "\n")
cat("INFERENTIAL STATISTICS\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

disease_0 <- df[df$target == 0, ]
disease_1 <- df[df$target == 1, ]

cat("\nT-Test: Comparing means between disease and no-disease groups\n")
for (col in numerical_cols) {
  if (col %in% names(df)) {
    group_0 <- disease_0[[col]]
    group_1 <- disease_1[[col]]
    
    if (length(group_0) > 0 && length(group_1) > 0) {
      test_result <- t.test(group_0, group_1)
      
      cat("\n", toupper(col), ":\n")
      cat("  t-statistic:", test_result$statistic, "\n")
      cat("  p-value:", test_result$p.value, "\n")
      cat("  Significant:", ifelse(test_result$p.value < 0.05, "Yes", "No"), "\n")
    }
  }
}

# Exploratory Analysis
cat("\n\n", paste(rep("=", 80), collapse = ""), "\n")
cat("EXPLORATORY DATA ANALYSIS\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

# Correlation matrix
numerical_cols_with_target <- c(numerical_cols, "target")
correlation_matrix <- cor(df[numerical_cols_with_target], use = "complete.obs")

cat("\nCorrelation with Target:\n")
target_corr <- correlation_matrix[, "target"]
target_corr <- target_corr[names(target_corr) != "target"]
print(sort(target_corr, decreasing = TRUE))

# Save correlation plot
png("../../results/correlation_matrix.png", width = 800, height = 600)
corrplot(correlation_matrix, method = "color", type = "upper", 
         order = "hclust", tl.cex = 0.8, tl.col = "black",
         addCoef.col = "black", number.cex = 0.7)
dev.off()

cat("\nAnalysis complete!\n")

